src/main/java/filter/expression/Qualifier.java

```diff
@@ -659,19 +659,7 @@ public static boolean hasUpdatedQualifier(FilterExpression expr) {
                 } else if (!bMilestone.isPresent()) {
                     return -1;
                 } else {
-                    Optional<LocalDate> aDueDate = aMilestone.get().getDueDate();
-                    Optional<LocalDate> bDueDate = bMilestone.get().getDueDate();
-
-                    if (!aDueDate.isPresent() && !bDueDate.isPresent()) {
-                        return 0;
-                    } else if (!aDueDate.isPresent()) {
-                        return 1;
-                    } else if (!bDueDate.isPresent()) {
-                        return -1;
-                    } else {
-                        return -(TurboMilestone.getDueDateComparator()
-                            .compare(aMilestone.get(), bMilestone.get()));
-                    }
+                    return -(TurboMilestone.getDueDateComparator().compare(aMilestone.get(), bMilestone.get()));
```

Comment:
So is that test failing? If not, is the the test is broken? If broken and related to this PR, fix it. If broken but not related to this PR, create a new issue for it.